rust/src/serialization.rs

```diff
@@ -556,11 +556,43 @@ impl DeserializeEmbeddedGroup for TransactionInput {
 
 impl cbor_event::se::Serialize for TransactionOutput {
     fn serialize<'se, W: Write>(&self, serializer: &'se mut Serializer<W>) -> cbor_event::Result<&'se mut Serializer<W>> {
-        serializer.write_array(cbor_event::Len::Len(if self.data_hash.is_some() { 3 } else { 2 }))?;
-        self.address.serialize(serializer)?;
-        self.amount.serialize(serializer)?;
-        if let Some(data_hash) = &self.data_hash {
-            data_hash.serialize(serializer)?;
+        match (&self.data, &self.script_ref) {
+            //post alonzo output
+            (Some(DataOption::Data(_)), _) | (_, Some(_)) => {
```

Comment:
Let's plz create functions (may be non-public) `.has_inline_data` and `.has_ref_script` and then make this check like:
```
if self.has_inline_data() || self.has_ref_script() {

} else {

}
```

Otherwise match looks confusing